### DISCUSSION TITLE
switch.controller_name? switch.program_name?

### DISCUSSION THREAD

**Discussion by brix29** [vinteo/hass-opensprinkler#147] [Q&A] [answered]
Hi,

First of all, thank you very much for this nice integration. Very much appreciated.
Currently I'm struggeling finding the entities switch.xyz_program (and switch.xyz_station)  in HA. 
I can find switch.xyz_program**_enabled** and binary_sensor.xyz_program**_running**.

Can you pls let me know, what I missed? 

Thank you very much!

Cheers, Brix

**Comment by vinteo** [comment, marked as answer]
Please read the readme on how to use services to switch the stations. Run
the services on the binary sensors
